modules/programs/firefoxpwa.nix

```diff
@@ -0,0 +1,254 @@
+                          '';
+                          example = {
+                            config.manifest_url = "https://developer.mozilla.org/manifest.f42880861b394dd4dc9b.json";
+                          };
+                        };
+                      };
+
+                      config = {
+                        desktopEntry.enable = lib.mkDefault isLinux;
+                        settings = {
+                          ulid = name;
+                          profile = profile.name;
+                          config = {
+                            name = config.name;
+                            document_url = config.url;
+                            manifest_url = config.manifestUrl;
+                          };
+                          manifest = {
+                            name = config.name;
+                            start_url = config.url;
+                          };
+                        };
+                      };
+                    }
+                  )
+                );
+              };
+              settings = lib.mkOption {
+                type = jsonFmt.type;
+                default = { };
+                description = ''
+                  Settings for this profile. See
+                  <https://github.com/filips123/PWAsForFirefox/blob/cb4fc76873cc51129d9290754768e6a340c521b2/native/src/components/profile.rs#L13-L34>
+                  for a list of available options.
+                '';
+              };
+            };
+
+            config.settings = {
+              ulid = name;
+              name = config.name;
+              sites = builtins.attrNames config.sites;
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions =
+      mkUlidAssertions [ "profiles" ]
+      ++ lib.concatMap (
+        name:
+        mkUlidAssertions [
+          "profiles"
+          name
+          "sites"
+        ]
+      ) (builtins.attrNames cfg.profiles)
+      ++ map (
+        name:
+        let
+          profiles = builtins.attrNames (lib.filterAttrs (_: profile: profile.sites ? ${name}) cfg.profiles);
+        in
+        {
+          assertion = builtins.length profiles == 1;
+          message = ''
+            Site with ULID '${name}' must be present in exactly one profile, but is present
+            in ${toString (builtins.length profiles)} profiles, namely ${
+              lib.concatMapStringsSep ", " (x: "'${x}'") profiles
+            }.
+          '';
+        }
+      ) (builtins.attrNames sites);
+
+    home.packages = lib.mkIf (cfg.package != null) [ cfg.package ];
+
+    xdg.dataFile."firefoxpwa/config.json" = lib.mkIf (cfg.settings != { }) {
+      text = lib.generators.toJSON { } cfg.settings;
+    };
+
+    xdg.desktopEntries = lib.mkMerge (
+      lib.mapAttrsToList (name: site: {
+        "FFPWA-${name}" = lib.mkIf site.desktopEntry.enable {
+          inherit (site.desktopEntry) icon categories;
```

Comment:
We should use `site.desktopEntry` as is so we can pass any parameter we want to it. For example: `class` so we can assign the proper WM class to the window and have a proper icon associated to it.